src/v/storage/parser_utils.cc

```diff
@@ -18,9 +18,21 @@
 #include "vlog.h"
 
 #include <seastar/core/byteorder.hh>
+#include <seastar/core/coroutine.hh>
 
 namespace storage::internal {
 
+ss::future<ss::stop_iteration>
+decompress_batch_consumer::operator()(model::record_batch& rb) {
+    _batches.push_back(
```

Comment:
don't you need to check if batch is compressed ? seems like easy to introduce a bug

Reply:
This is already performed by the function itself

Reply:
smart.